src/mdev.rs

```
@@ -123,6 +123,31 @@ impl<'a> MDev<'a> {
         Ok(())
     }
 
+    pub fn add_attributes(&mut self, attrs: &serde_json::Value) -> Result<()> {
+        if let Some(attrarray) = attrs.as_array() {
```

Now that we've exposed this function to be used by other callers, perhaps we should return an error if the json value is not an array.